docs/adrs/adr-004-nix-flakes.md

```diff
@@ -0,0 +1,44 @@
+# ADR 4: Adopting Nix Flakes
+
+## Context
+
+We're currently using Nix flakes, which has lead to some friction with adoption - users new to Nix see an added complexity. There has been a call to simplify with refactoring Flakes into Nix expressions.
+
+### tl;dr Points
+
+Points pro de-flaking:
+- Easier adoption, less friction, the ability to "dip feet" vs "dive-in" with impure shells.
```

Comment:
Would perhaps adding "--impure" switch in our setup, help with adoption?

Reply:
Yup, awesome work on https://github.com/Qarik-Group/disruptor/pull/23.